### DISCUSSION TITLE
Data only refeshs after reloading the page

### DISCUSSION THREAD

**Discussion by PeetThesing** [neuefische/questions#790] [Web] [answered]
Hi there,

we encountered a problem: Our data only updates after a complete reload of the page.
It doesn't reload after adding a snippet or deleting one.

We think this problem exists since the implementation of the bookmarks and moving the data fetching to the _app.js.

Here is our[ PR](https://github.com/PeetThesing/capstone-project_snippet-manager/pull/56)

We tried using mutate, but that doesn't seem to help.

We appreciate any help.

**Comment by NikRoe** [comment]
Hi there,

I got it working by calling `mutate()` in your `_app.js`, but I don't like that solution very much.

I'll talk about this with @onemanwenttomow soonish and then we'll come back to you.

**Reply by NikRoe** [reply, marked as answer]
Hi there,

the one and only @onemanwenttomow was able to find a better solution!

Your approach with using `mutate` was the absolute right one, however, you need to make a very small adjustment to it:
in your `handleDelete` in `SnippetDetailsPage` you need to add a slash to the string.
```js
mutate(`/api/snippets`);
```
The reason behind this is that this string needs to be identical to the URL string of your fetch in the `_app.js` to actually case swr to revalidate the data.

In your `createSnippet` in `FormPage` you can call `mutate()` without passing an argument to it.
Since you destructure it from the return of `useSWR("/api/snippets");` it will always revalidate the URL you specified here.

Hope this already makes it work on your end as intended 🙌

**Reply by PeetThesing** [reply]
Thank you!!
We added the slash in the _app.js, and now it finally works again!
Such a small change but sweeping consequences...
